### PR TITLE
EES-1145 Make `Release.PreviousVersionId` field nullable

### PR DESCRIPTION
**Pull request by ntsim**
This PR changes `Release.PreviousVersionId` to nullable so that we can make it easier to bulk delete releases when deleting themes/topics. Currently `PreviousVersionId` cannot be nulled and EF is consequently unable to delete all of a theme/topic's releases in a single operation due to the self-referential nature of the `Release` -> `PreviousVersion` relationship.

Additionally, `PreviousVersionId` can take on non-sensical IDs such as `Guid.Empty` or its own `Id` when it is the first release (doesn't make sense as the previous version shouldn't be itself). These changes should correct the semantics of `PreviousVersion` and `PreviousVersionId` i.e. make them properly optional.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentService.cs
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentService.cs:52]: You could move this within the `if` above since that's also checking whether a Previous Version exists

**Review by benoutram — APPROVED** [review]
This all seems sensible, but @shakes90 was talking about this yesterday so I think it would also be good to wait for his approval.

File: src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentService.cs
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentService.cs]: Cool done

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200825153306_EES1145AllowReleasePreviousVersionIdToBeNullable.cs
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200825153306_EES1145AllowReleasePreviousVersionIdToBeNullable.cs:27]: With bug https://dfedigital.atlassian.net/browse/EES-1344 we are going to lose track of the Amendments and whether they are a previous version or not, if we run this before we fix that.

**Review by benoutram — CHANGES_REQUESTED** [review]

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200825153306_EES1145AllowReleasePreviousVersionIdToBeNullable.cs
Comment (benoutram) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200825153306_EES1145AllowReleasePreviousVersionIdToBeNullable.cs:27]: Actually, not sure there are any Releases with PreviousVersionId equal to 00000000-0000-0000-0000-000000000000 in the Content database, I think they are all in the Statistics databases.

**Review by benoutram — APPROVED** [review]

**Pull request merged**
